sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/CoverageTest.java

```diff
@@ -161,47 +161,83 @@ public void givenSingleBranchPointPerLine_getBranchCoverage_returnsEmpty() {
     final String filePath = "filePath";
 
     Coverage sut = new Coverage();
-    sut.add(new BranchPoint(filePath, 1, 2, 3, 4));
-    sut.add(new BranchPoint(filePath, 2, 5, 6, 7));
+    sut.add(new BranchPoint(filePath, 1, 2, 3, 4, 5));
+    sut.add(new BranchPoint(filePath, 2, 6, 7, 8, 9));
 
     assertThat(sut.getBranchCoverage(filePath)).isEmpty();
   }
 
+  @Test
+  public void branchPointsMerging() {
+    final String filePath = "filePath";
+
+    Coverage sut = new Coverage();
+
+    // Identical branch points are merged
+    sut.add(new BranchPoint(filePath, 1, 1, 2, 0, 1));
+    sut.add(new BranchPoint(filePath, 1, 1, 2, 0, 1));
+
+    // Branch points with different line are not merged
+    sut.add(new BranchPoint(filePath, 2, 1, 2, 0, 1));
+    sut.add(new BranchPoint(filePath, 3, 1, 2, 0, 1));
+
+    // Branch points with different offset are not merged
```

Comment:
uhm, I see below that lines `4` , `5` and `6` do get merged (and the comment says "are not merged")

Reply:
`// Branch points with different offset are correctly aggregated`